src/main/scala/org/clulab/asist/agents/DialogAgentReprocessor.scala

```diff
@@ -191,43 +181,52 @@ class DialogAgentReprocessor (
       // report status
       stateReport(s)
 
-      // if we have another file, run it
-      if(s.fileInfoIterator.hasNext) {
-        // next file iteration
-        val fileInfo = s.fileInfoIterator.next
+      fileIteration(s)
+    }
+  }
 
-        val inputFileName = fileInfo._1
-        val inputFileLines = fileInfo._2
-        val outputFileName = ta3FileName(inputFileName)
+  def fileIteration(s: RunState): Unit = {
 
-        val newState = s.copy(
-          lineIterator = LocalFileUtils.lineIterator(inputFileName),
-          fileWriter = Some(new PrintWriter(new File(outputFileName))),
-        )
-        val theKingsEnglish = if(inputFileLines == 1) "line" else "lines"
+    // if we have another file, run it
+    if(s.fileInfoIterator.hasNext) {
+      // next file iteration
+      val fileInfo = s.fileInfoIterator.next
 
-        logger.info(
-          "Reading {} {} from {} ...", 
-          inputFileLines, 
-          theKingsEnglish, 
-          inputFileName
-        )
+      val inputFileName = fileInfo._1
+      val inputFileLines = fileInfo._2
+      val outputFileName = ta3FileName(inputFileName)
 
-        dacClient match {
-          case Some(dc: DacClient) => 
-            dc.resetServer(newState)
-          case None =>
-            futureIteration(newState)
-        }
-      }
-      else {
-        // done
-        logger.info("All file processing has finished.")
-        finalReport(s)
-        system.terminate()
-        dacClient.foreach(dc => dc.shutdown)
+      val newState = s.copy(
+        lineIterator = LocalFileUtils.lineIterator(inputFileName),
+        fileWriter = Some(new PrintWriter(new File(outputFileName))),
+      )
+      val theKingsEnglish = if(inputFileLines == 1) "line" else "lines"
+
+      logger.info(
```

Comment:
same comment as a few other places

Reply:
Scala lacks a ternary operator so I kind of made one up.